teuthology/orchestra/remote.py

```diff
@@ -599,6 +598,22 @@ def sudo_write_file(self, path, data, **kwargs):
         """
         self.write_file(path, data, sudo=True, **kwargs)
 
+    def write_temp_file(self, **kwargs):
+        """
+        This is a convenience method that call write_file() but without sudo.
+        Reason for excluding sudo is that files in /tmp can not be written by
+        any user (including the root user) besides the file owner.
+
+        Takes arguments of write_file() and of mktemp().
+        """
+        kwargs['sudo'] = False
+
+        if not kwargs.get('path'):
+            kwargs['path'] = self.mktemp(kwargs.pop('suffix',  None),
+                                         kwargs.pop('parentdir', None))
+        self.write_file(**kwargs)
```

Comment:
Above fixes are because these two parameters are not valid for this call.

Reply:
Looking at the code snippet I am confused. Which call? Call to `write_file()`? `suffix` and `parentdir` will not be contained by `kwargs` at this point because `pop()` will remove them from kwargs.

Reply:
If `kwargs['path'] is None` then `kwargs['suffix']` and `kwargs['parentdir']` will be left set, causing an error in the call to `self.write_file`.